### PR TITLE
[eas-build-job] add new iOS builder image

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Add new iOS builder image for XCode 14

# How

Add new iOS builder image for XCode 14

# Test Plan

Tests

### CODE REVIEW

File: packages/build-tools/src/ios/pod.ts [hunk 1]
```diff
@@ -9,28 +8,6 @@ import { BuildContext } from '../context';
 export async function installPods<TJob extends Ios.Job>(ctx: BuildContext<TJob>): Promise<void> {
   const iosDir = path.join(ctx.reactNativeProjectDirectory, 'ios');
 
-  if (ctx.env.EAS_BUILD_COCOAPODS_CACHE_URL) {
```
Comment (wkozyra95) [inline packages/build-tools/src/ios/pod.ts:12]: Do not include commits from other reviews, like that. If you want to have stacked PR then this PR should point to the branch where base commit is from, but in this case there is no need for that because changes are not related to each other

**Review by wkozyra95 — APPROVED** [review]

File: packages/build-tools/src/ios/pod.ts (hunk 1 above)
Comment (szdziedzic) [reply packages/build-tools/src/ios/pod.ts:12]: Sorry, I accidentally did it, it wasn't intentional action

**Review by dsokal — APPROVED** [review]
